spec/amber/router/context_spec.cr

```diff
@@ -267,4 +267,14 @@ describe HTTP::Server::Context do
       end
     end
   end
+
+  describe "X-Powered-By Header" do
+    it "initializes context with X-Powered-By: Amber" do
+      headers = HTTP::Headers.new
+      request = HTTP::Request.new("GET", "/", headers)
+      context = create_context(request)
+      context.response.headers.add "X-Powered-By", "Amber"
```

Comment:
In order to test if the pipeline is correctly adding the header, you shouldn't set this in the test.  you should let the pipeline add it.

Reply:
Ok, let me fix this

Reply:
Try something like this:
```
pipeline = Amber::Pipe::Pipeline.new
pipeline.call(context)
context.response.headers["X-Powered-By"].should eq "Amber"
```

Reply:
@faustinoaq @drujensen is correct you can unit test the Pipeline and verify that the header is set. That's a simple test setup.